receiver/filebeatreceiver/filebeat_to_logdata.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filebeatreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filebeatreceiver"
+
+import (
+	"errors"
+	"sort"
+	"time"
+
+	"github.com/elastic/go-lumber/lj"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+const (
+	cannotConvertValue = "cannot convert field value to attribute"
+)
+
+// FilebeatToLogData transforms Filebeat events into logs
+func FilebeatToLogData(logger *zap.Logger, batch *lj.Batch) (plog.Logs, error) {
+	ld := plog.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
```

Comment:
Are we always adding `LogRecord` to the same `ResourceLogs`? Meaning we are assuming all the logs coming from same "resource" / source?

According to OTLP logs model/proto: https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto#L44
> message LogsData {
>  // An array of ResourceLogs.
>  // For data coming from a single resource this array will typically contain
>  // one element. Intermediary nodes that receive data from multiple origins
>  // typically batch the data before forwarding further and in that case this
>  // array will contain multiple elements.
>  repeated ResourceLogs resource_logs = 1;
>}

Reply:
Just to capture our discussion here.  This can be done in two ways.
1. Allow the receiver to just act as a receiver and not bother about resource attribute movements.  This way the receiver's responsibility is to receive the data and transmit in OTLP intermediate pdata downstream.  The downstream processors like `groupbyattrs` can take care of the resource attribute movement. 
2. Make changes to the receiver to support a configuration to allow resource attribute movement as part of the receiver itself.
Functionally, both should work because all of these are intermediate stages in the pipeline and the final expectation of the complete LogsData as per the logs.proto is only at the output of the final stage of the pipeline otherwise between stages its just the intermediate structure.  
That said, still as you mentioned about the memory churn that could happen between stages due to the immutable nature of the intermediate structure we may still have to pursue the 2nd approach depending on the Perf testing.
But, atleast as a first PR I suppose this should address the functionality, we could improve the performance in subsequent versions according to my opinion.